experiments/tf_trainer/common/model_trainer.py

```diff
@@ -191,29 +192,62 @@ def __init__(self, dataset: ds.DatasetInput,
     self._model = model
     self._estimator = model.estimator(self._model_dir())
 
-  def train_with_eval(self):
-    """Train with periodic evaluation."""
+  def train_with_eval(self,
+                      serving_input_receiver_fn=None,
+                      example_key_name=None):
+    """Train with periodic evaluation.
+
+    Args:
+      serving_input_receiver_fn: Input serving function.
+      example_key_name: Name for example_key feature (needed for eval).
+    """
+    if example_key_name:
+      self._estimator = self._add_estimator_key(self._estimator, example_key_name)
+      self._example_key_name = example_key_name
+
     training_hooks = None
     if FLAGS.enable_profiling:
       training_hooks = [
           tf.train.ProfilerHook(
               save_steps=10,
-              output_dir=os.path.join(self._model_dir(), 'profiler'))
+              output_dir=os.path.join(self._model_dir(), 'profiler')),
       ]
 
     train_spec = tf.estimator.TrainSpec(
         input_fn=self._dataset.train_input_fn,
         max_steps=FLAGS.train_steps,
         hooks=training_hooks)
+
+    exporter = None
+    if FLAGS.n_export == -1:
+      if not serving_input_receiver_fn:
```

Comment:
if serving_input_receiver_fn is None, what about not exporting the model instead of raising error ? Sometimes, you want to train without thinking about a serving_fn.